src/LinkPreview/Parser/YoutubeParser.php

```diff
@@ -116,7 +116,8 @@ public function parseLink()
     {
         $this->readLink();
         $link = $this->getLink();
-        $htmlData = $this->parseHtml($link->getContent());
+        $content = $link->getCharset() == 'utf-8' ? utf8_decode($link->getContent()) : $link->getContent();
```

Comment:
comparison needs to be case-insensitive.

Example: Vimeo sends header with ``Content-Type:text/html; charset=UTF-8``